source/iterator/iterator.go

```diff
@@ -78,13 +81,16 @@ func (c *SheetsIterator) startIterator(ctx context.Context) func() error {
 			case <-c.ticker.C:
 				records, err := c.sheets.GetSheetRecords(ctx, c.rowOffset)
 				if err != nil {
+					// haris: we should be wrapping the errors, to provide more context about
+					// how it came to the error (e.g. fmt.Errorf("failed fetching records at offset %v: %w"))
 					return err
 				}
 				if len(records) == 0 {
 					continue
 				}
 				select {
 				case c.caches <- records:
+					// haris: are the records guaranteed to be sorted?
```

Comment:
Yes, we get the rows sorted by row id from google sheets and process them sequentially.

Reply:
Thanks for the clarification :+1: